### DISCUSSION TITLE
Offline mode needed!

### DISCUSSION THREAD

**Discussion by dskudi** [dani-garcia/vaultwarden#5254] [Ideas]
Would love if I could create passwords and notes and other items while not having an Internet connection. That the new items could be saved in the local app memory until I have an Internet connection again and then it would sync to my online vault. Is this possible now? Am I missing something?

I use Vaultwarden primarily through the Bitwarden iPhone app. But I also have it sync with my iPad as well as my Mac and PC computers. Would love the off-line mode capability on the iPhone app in particular.

**Comment by BlackDex** [comment]
This is a client side item of course, but as far as i know they all support this already.
You just need to make sure a reverse proxy doesn't respond with an error code in some way.

But you can put your phone into Flightmode and still work with the stored data.
You can't however add new items in offline mode, as is also stated here on the Bitwarden site https://bitwarden.com/help/using-bitwarden-offline/

**Reply by dskudi** [reply]
Yeah, I realized after posting this that it's an app issue. I know that I can view things off-line, but would love if I could edit and add new items while off-line and it would just be saved locally until I was back online again for it to sync up. I had one password before this, and it worked perfectly for these type of situations. Really wish that Bitwarden had this feature, as well as it comes in handy when creating secure notes and other edits while not able to sync up to my server.

Appreciate the comment!

**Comment by tessus** [comment]
> You just need to make sure a reverse proxy doesn't respond with an error code in some way.

@BlackDex can you please clarify that? If the backend service is not available, a reverse proxy always returns a 503 status code. Do we have to configure the reverse proxy to return 200 instead? Somehow this makes no sense, because even with a return code of 200, the app won't be available/accessible.
None of the reverse-proxy examples in the wiki have such a directive either, thus I am a bit puzzled by your statement.

**Reply by jjlin** [reply]
See https://vaultwarden.discourse.group/t/offline-online-access/2298/2

**Reply by BlackDex** [reply]
True. But that is inherent on reverse proxies.
I'm not sure what i should explain further, I'm stating what the clients do upon a error code.
I do not think it's even tested at Bitwarden, since the self-hosted version is configured the same, and i guess the cloud version will probably do something similar, and then those clients will then logout too.

**Reply by tessus** [reply]
hmm, according to the link jjlin sent, all is good with 503 (no logout), unless I misread the code.

**Reply by BlackDex** [reply]
That is only for the none mobile clients. Not sure how this is for the (new) mobile clients which do not use those libraries.

**Reply by tessus** [reply]
I would hope they behave consistently, although from previous experience that might not be the case. bw is rather inconsistent in their code.

**Comment by horst789github** [comment]
any news here? i have the problem on the Windows PC, if i connected to Work VPN "my" vaultwarden wont open(loads very long) on PC, because cant reach my vaultwarden server.

**Comment by JOduMonT** [comment]
I was expecting to be able to use my BitWarden Android Clients, and others while I'm offline

But when my server is down 
The BitWarden Client on Android
After authentication; it simply close without loading the interface.

I believe it would be wonderful,
imagine a distopian future were part of internet is temporary shutdown; and or DNS routes are poisoned; that means you lose access to your password,

**Reply by kpfleming** [reply]
If you log out on the client then it has to contact the server to complete the login process. If you only *lock* the client and don't log out, then you can unlock even if the server is unreachable. The client cannot handle a login on its own, it requires the server's assistance.

**Reply by JOduMonT** [reply]
Interestingly;

via the browser autocomplete, I can login in my vault and search for password; but not via the android app it self

Both are lock, not logged out

**Reply by JOduMonT** [reply]
@kpfleming I just tested your lock out vs log out
and what you mentionned is true with my vaultwarden via the browser plugin
and now also true on my android ;)
so nothing to complain anymore :P
maybe the update fixed it.
